pkg/controller/node/nodecontroller.go

```diff
@@ -193,12 +202,32 @@ func NewNodeController(
 	return nc
 }
 
+// generateNewAvailabeCIDRs does generating new Available CIDRs index and insert
+// them into availableCIDRs set. Everytime it will generate 256 new CIDRs,
+// once there are no more CIDRs in the network, return false
+func (nc *NodeController) generateNewAvailabeCIDRs() bool {
+	if nc.firstAvailable+256 > nc.maxCIDRs {
```

Comment:
This is checking if we do not have any available CIDRs which has not being generated(or inserted into availbleCIDRs set). maxCIDRs represent the maximum number of CIDRs we have in our system(based on clusterCIDR *net.IPNet). Checking nc.firstAvailable+256 > nc.maxCIDRs is because everytime we call generateNewAvailabeCIDRs(), we will generate 256 CIDRs, we want to make sure they are enough CIDRs have not been generated yet.